test/sqllogictest/mz_introspection_index_accounting.slt

```diff
@@ -0,0 +1,478 @@
+mz_message_counts_received_raw_s2_primary_idx  CREATE␠INDEX␠"mz_message_counts_received_raw_s2_primary_idx"␠IN␠CLUSTER␠[s2]␠ON␠"mz_internal"."mz_message_counts_received_raw"␠("channel_id",␠"from_worker_id",␠"to_worker_id")
+mz_arrangement_batcher_allocations_raw_s2_primary_idx  CREATE␠INDEX␠"mz_arrangement_batcher_allocations_raw_s2_primary_idx"␠IN␠CLUSTER␠[s2]␠ON␠"mz_internal"."mz_arrangement_batcher_allocations_raw"␠("operator_id",␠"worker_id")
+mz_compute_delays_histogram_raw_s2_primary_idx  CREATE␠INDEX␠"mz_compute_delays_histogram_raw_s2_primary_idx"␠IN␠CLUSTER␠[s2]␠ON␠"mz_internal"."mz_compute_delays_histogram_raw"␠("export_id",␠"import_id",␠"worker_id",␠"delay_ns")
+mz_message_batch_counts_sent_raw_s2_primary_idx  CREATE␠INDEX␠"mz_message_batch_counts_sent_raw_s2_primary_idx"␠IN␠CLUSTER␠[s2]␠ON␠"mz_internal"."mz_message_batch_counts_sent_raw"␠("channel_id",␠"from_worker_id",␠"to_worker_id")
+mz_scheduling_parks_histogram_raw_s2_primary_idx  CREATE␠INDEX␠"mz_scheduling_parks_histogram_raw_s2_primary_idx"␠IN␠CLUSTER␠[s2]␠ON␠"mz_internal"."mz_scheduling_parks_histogram_raw"␠("worker_id",␠"slept_for_ns",␠"requested_ns")
+mz_compute_import_frontiers_per_worker_s2_primary_idx  CREATE␠INDEX␠"mz_compute_import_frontiers_per_worker_s2_primary_idx"␠IN␠CLUSTER␠[s2]␠ON␠"mz_internal"."mz_compute_import_frontiers_per_worker"␠("export_id",␠"import_id",␠"worker_id")
+mz_message_batch_counts_received_raw_s2_primary_idx  CREATE␠INDEX␠"mz_message_batch_counts_received_raw_s2_primary_idx"␠IN␠CLUSTER␠[s2]␠ON␠"mz_internal"."mz_message_batch_counts_received_raw"␠("channel_id",␠"from_worker_id",␠"to_worker_id")
+mz_dataflow_shutdown_durations_histogram_raw_s2_primary_idx  CREATE␠INDEX␠"mz_dataflow_shutdown_durations_histogram_raw_s2_primary_idx"␠IN␠CLUSTER␠[s2]␠ON␠"mz_internal"."mz_dataflow_shutdown_durations_histogram_raw"␠("worker_id",␠"duration_ns")
+mz_compute_operator_durations_histogram_raw_s2_primary_idx  CREATE␠INDEX␠"mz_compute_operator_durations_histogram_raw_s2_primary_idx"␠IN␠CLUSTER␠[s2]␠ON␠"mz_internal"."mz_compute_operator_durations_histogram_raw"␠("id",␠"worker_id",␠"duration_ns")
+mz_dataflow_operator_reachability_raw_s2_primary_idx  CREATE␠INDEX␠"mz_dataflow_operator_reachability_raw_s2_primary_idx"␠IN␠CLUSTER␠[s2]␠ON␠"mz_internal"."mz_dataflow_operator_reachability_raw"␠("address",␠"port",␠"worker_id",␠"update_type",␠"time")
+
+# Record all transitive dependencies (tables, sources, views, mvs) of indexes on
+# the mz_introspection cluster.
+# Include each column of each dependent object, so we can detect changes to
+# object definitions that may impact the memory utilization of the index.
+query TT
+SELECT DISTINCT o.name dependent_object_name, co.name column_name
+FROM mz_indexes i
+LEFT JOIN mz_internal.mz_object_transitive_dependencies d ON i.id = d.object_id
+LEFT JOIN mz_objects o ON o.id = d.referenced_object_id
+JOIN mz_columns co on co.id = o.id
+WHERE cluster_id =
+  (SELECT id FROM mz_clusters WHERE name = 'mz_introspection')
```

Comment:
Ditto. It's funny; you can actually see that the arbitrary order chooses to sort first by length and then lexicographically.

Reply:
Hah you're right, good call!